concepts/conditionals/about.md

```diff
@@ -0,0 +1,178 @@
+if (number == 0)
+    return "zero";
+else if (number < 0)
+    return "negative";
+else if (number < 10)
+    return "single digit";   
+else if (number % 2 == 0)
+    return "even";
+else
+    return "odd";
+```
+
+## Switch
+
+
+Sometimes an `if-else-if ladder` can be better represented by a
+`switch`. Like `if`, the expression being tested is always put
+in parentheses.
+
+```c
+switch (number) {
+    case -1: return "negative one";
+    case 0: return "zero";
+    case 1: return "positive one";
+    case 10: return "positive ten";
+    case -10: return "negative ten";
+    default: return "don't care";
+}
+```
+
+The `default` case executes if the value being tested does not
+match any of the previous cases.
+
+If a `switch` case is not returning then a `break` is usually
+inserted to end each case, otherwise the logic "falls through"
+to the next case. Sometimes fallthrough is used
+intentionally
+
+```c
+switch (number) {
+    case 0: msg = "zero"; break;
+    case 1: msg = "one"; break;
+    case 2: case 3: case 4:
+        msg = "more than one and less than five"; break;
+    default: msg = "don't care";
+}
+```
+
+When fallthrough happens unintentionally it's a bug.
+
+```c
+switch (number) {
+    case 0: msg = "zero"; // bug! fallthrough of 0 will set msg to "one"
+    case 1: msg = "one"; break; 
+    case 2: case 3: case 4:
+        msg = "more than one and less than five"; break;
+    default: msg = "don't care";
+}
+```
+
+A switch can be beneficial when testing against specific values
+instead of a range of values. It can be faster than an
+`if-else-if ladder` because the compiler may be able to optimize
+the `switch` statement in ways it can't do with a ladder. A switch
+can sometimes look more readable than a ladder.
+
+## Ternary (aka Conditional) Operator
+
+The ternary operator is a three-part conditional statement that is
+often used for setting a value. The expression to be tested is
+before the question mark and does not have to be in parentheses.
+The value returned if true is before the colon, the value returned
+if false is after the colon. Either a value literal or a variable
+can be returned.
+
+```c
+int i = j > 5 ? 5: j;
+```
+
+Ternary operators can be nested to reduce syntactic clutter.
+
+```c
+// the below could also be 
+i = j > 5 ? j :k > 5 ? k: 10;
+// but that may not be quite as readable
+
+i = j > 5 ? j
+    :k > 5 ? k
+    :10;
+
+// which is the same as
+if (j > 5) i = j;
+else if (k > 5) i = k;
+else i = 10;
+```
+
+## Conditional Comparisions for Floating Point Values
+
+Because of how floating point numbers are implemented, a simple math
+operation may result in a rounding error which can cause a test for
+equality to not return the expected result.
+
+```c
+    a = 1.345f;
+    b = 1.123f;
+    c = a + b;
+    // c may equal something like 2.4679999352
+    if (c == 2.468)
+        return "I won the lottery!";
+```
+
+A tolerance value, referred to as `epsilon`, can be used to determine a range
```

Comment:
Could there be an example of using an epsilion value in an if statement?
If this paragraph came before the above example, then probably that example could contain both the existing usage and an epsilion usage?